### PR TITLE
More doc

### PR DESCRIPTION
**Pull request by yifuwang**
As title.

### CODE REVIEW

File: docs/source/getting_started.rst [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+    # In the same process or in another process
+    snapshot.restore(app_state=app_state)
+    # global RNG state => {x}
+
+
+Elasticity (experimental)
+-------------------------
+
+Distributed applications can restore from a snapshot taken with a different world size as long as the snapshot only contains **replicated object** or **sharded object**:
```
Comment (edward-io) [inline docs/source/getting_started.rst:145]: s/object/objects

File: docs/source/getting_started.rst [hunk 2]
```diff
@@ -0,0 +1,152 @@
+Getting started
+***************
+
+
+:class:`~torchsnapshot.Snapshot` is the core API of TorchSnapshot. The class represents application state persisted in storage. A user can take a snapshot of an application via :meth:`Snapshot.take() <torchsnapshot.Snapshot.take>` (a.k.a saving a checkpoint), or restore the state of an application from a snapshot via :meth:`Snapshot.restore() <torchsnapshot.Snapshot.restore>` (a.k.a loading a checkpoint).
```
Comment (edward-io) [inline docs/source/getting_started.rst:5]: i think we should specify application = training / inference script somewhere

File: docs/source/getting_started.rst [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+
+.. _app-state:
+
+
+Describing the application state
+--------------------------------
+
+Before using :class:`~torchsnapshot.Snapshot` to save or restore application state, the user needs to **describe the application state**. This is done by creating a dictionary that contains all **stateful objects** that the user wishes to capture as application state. Any object that exposes ``.state_dict()`` and ``.load_state_dict()`` are considered stateful objects. Common PyTorch objects such as :class:`~torch.nn.Module`, :class:`~torch.optim.Optimizer`, and LR schedulers all qualify as stateful objects and can be captured directly. Objects that don't meet this requirement can be captured via :class:`~torchsnapshot.StateDict`.
```
Comment (edward-io) [inline docs/source/getting_started.rst:14]: do you want to mention and link the Stateful protocol here?

**Review by edward-io — APPROVED** [review]
lgtm, some nits

File: docs/source/getting_started.rst (hunk 3 above)
Comment (yifuwang) [reply docs/source/getting_started.rst:14]: I want to elaborate it in "API Reference" to make "Getting Started" more concise.

**Pull request closed without merging**